### PR TITLE
server: optionally disable update checks

### LINKED ISSUE #6212: server: disable update check on circleci
Our registration stats are being skewed by the "clusters" run in our CI test suite. It looks like 10-12 tests run long enough to complete a registration in a typical run. We should disable this by either special-casing circleci or delaying the first ping until the cluster has been up for a few minutes.

### PR DESCRIPTION
**Pull request by dt**
Having our CI runs checking for updates isn't useful, and some deployments
may want to disable the check anyway. This also disables the usage-stats
reporting by that node (even if the cluster has it enabled).

Fixes #6212.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/6302)

### CODE REVIEW

**Comment by bdarnell** [comment]
:lgtm:

---

Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[server/updates.go, line 72 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6302#-KGESmYjnuc56kuw42N8:-KGESmYkhVbYh3ZxgFtd:-1332787189) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8f384bf57ecf79258d26f0e539c6421bd7a8bee/server/updates.go#L72)):*
Remove "except in tests"? This method is clearly called in at least some tests.

---

*[server/updates.go, line 75 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6302#-KGESwU-8yj1A-cyi0b7:-KGESwU-8yj1A-cyi0b8:590563051) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8f384bf57ecf79258d26f0e539c6421bd7a8bee/server/updates.go#L75)):*
I wouldn't log anything here. We don't need to log this in every test.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6302#-:-KGET1qa9ivb8gr2Wp9A:1417088721)*

**Comment by dt** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[server/updates.go, line 72 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6302#-KGESmYjnuc56kuw42N8:-KGI5a-2IXj8ZrFPi0H2:2029136546) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8f384bf57ecf79258d26f0e539c6421bd7a8bee/server/updates.go#L72)):*
Expanded a bit.

---

*[server/updates.go, line 75 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6302#-KGESwU-8yj1A-cyi0b7:-KGI5dLSH2_IbT2njur3:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/a8f384bf57ecf79258d26f0e539c6421bd7a8bee/server/updates.go#L75)):*
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6302)*

File: server/updates.go [hunk 1]
```diff
@@ -67,9 +68,16 @@ type storeInfo struct {
...
 }
 
-// SetupReportingURLs parses the phone-home for version updates URL and should be
-// called before server starts except in tests.
+// SetupReportingURLs parses the phone-home for version updates URL and should
+// be called before a server starts.
+// Where update checks are not useful (eg in tests), skipping this call, or
+// setting env var COCKROACH_SKIP_UPDATE_CHECK=1, skips the acutal network calls
+// (note though the check is treated as having succeeded, meaning the cluster
+// will wait ~24 to check again).
```
Comment (tschottdorf) [inline server/updates.go]: s/~24/~24h/?

Comment (tschottdorf) [inline server/updates.go]: or even just s/~24//

**Comment by dt** [comment]
Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[server/updates.go, line 76 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6302#-KGES3j_x-wj_48c1lws-r2-76:-KGIFdfelvVwBqQh11-_:-907695340) ([raw file](https://github.com/cockroachdb/cockroach/blob/b530e1bb3242d9507d8182b8b71e84ca4ee45695/server/updates.go#L76)):*
Done (ish).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6302)*

**Pull request merged**